src/lib/decorators.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+/**
+ * Class decorator that defines the decorated class a custom element.
+ *
+ * @param tagName The name of the custom element to define.
+ */
+export const customElement = (tagName: string) => (clazz: any) => {
```

Comment:
Have you considered how this will extend to scoped custom element registries?

Reply:
Not very thoroughly, but this is basically the same as a `customElements.define()` statement in the module. That will define on the global registry, and redefinitions of the tag name in scoped registries should be ok.